src/main/java/com/spotify/docker/client/LogReader.java

```diff
@@ -77,4 +91,49 @@ public void close() throws IOException {
     // where close is a no-op.
     copy(stream, nullOutputStream());
   }
+
+  protected synchronized int readWithTimeout(final InputStream in, final byte[] buff,
```

Comment:
why make this synchronized? because you are re-assigning the readStreamExecutorService field?